### ISSUE TITLE
Check invalid combinations of --ci, --format, and --json

### ISSUE THREAD

**Issue by userzimmermann** [coala/coala#4055]
While reviewing https://github.com/coala/coala/pull/4036 , I realized that any combinations of `--ci`, `--format`, and `--json` are allowed for `coala` without any error. That's confusing. There is just a precedence for applying one of those flags and just ignoring the others:

```
--json
--format
--ci
```

`coala --ci --format` and `coala --ci --json` might be ok, but `coala --format --json` - which just ignores `--format` - definitely not.

@MariosPanag Interested? :)

**CC** @sils

**Comment by userzimmermann** [comment]
@Techievena This should be aligned with https://github.com/coala/coala/pull/3750 from @Nosferatul

**Comment by Nosferatul** [comment]
So, we want to warn users about using a combination of --ci, --format, and --json? @userzimmermann

**Comment by userzimmermann** [comment]
@Nosferatul Yes. Just like you did in https://github.com/coala/coala/pull/3750/files#diff-bad7250907d1790b012db412321e54c2R125 for `coala --ci --format`, whereby I'm not sure if `--ci --format` and `--ci --json` combinations are really worth an error, since `--format` and `--json` are non-interactive themselves. That needs to be discussed. But `--format --json` combination definitely needs that kind of error.

**Comment by Nosferatul** [comment]
well I could add that in my pr if that's ok, with you and @Techievena

**Comment by Techievena** [comment]
Yeah that's completely fine with me @Nosferatul You may add to your PR because you already have an open PR this stuff can happen much faster but the issue you're working upon will not be related to the PR.
